docs/usage.rst

```diff
@@ -0,0 +1,134 @@
+..
+    Copyright (c) 2022-2023 ETH Zurich
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _usage:
+
+=====
+Usage
+=====
+
+.. _getting_started:
+
+Getting started
+===============
+
+The recommended way to install pika is through `spack <https://spack.readthedocs.io>`_:
+
+.. code-block:: bash
+
+   spack install pika
+
+See
+
+.. code-block:: bash
+
+   spack info pika
+
+for available options.
+
+Manual installation
+-------------------
+
+If you'd like to build pika manually you will need CMake 3.22.0 or greater and a recent C++ compiler
+supporting C++17:
+
+- `GCC <https://gcc.gnu.org>`_ 9 or greater
+- `clang <https://clang.llvm.org>`_ 11 or greater
+
+Additionally, pika depends on:
+
+- `header-only Boost <https://boost.org>`_ 1.71.0 or greater
+- `hwloc <https://www-lb.open-mpi.org/projects/hwloc/>`_ 1.11.5 or greater
+- `fmt <https://fmt.dev/latest/index.html>`_ 9.0.0 or greater
+
+pika optionally depends on:
+
+* `gperftools/tcmalloc <https://github.com/gperftools/gperftools>`_, `jemalloc
+  <http://jemalloc.net/>`_, or `mimalloc <https://github.com/microsoft/mimalloc>`_. It is *highly*
+  recommended to use one of these allocators as they perform significantly better than the system
+  allocators. You can set the allocator through the CMake variable ``PIKA_WITH_MALLOC``. If you want
+  to use the system allocator (e.g. for debugging) you can do so by setting
+  ``PIKA_WITH_MALLOC=system``.
+* `CUDA <https://docs.nvidia.com/cuda/>`_ 11.0 or greater. CUDA support can be enabled with
+  ``PIKA_WITH_CUDA=ON``. pika can also be built with nvc++ from the `NVIDIA HPC SDK
+  <https://developer.nvidia.com/hpc-sdk>`_. In the latter case, set ``CMAKE_CXX_COMPILER`` to
+  ``nvc++``.
+* `HIP <https://rocmdocs.amd.com/en/latest/index.html>`_ 5.2.0 or greater. HIP support can be
+  enabled with ``PIKA_WITH_HIP=ON``.
+* `MPI <https://www.mpi-forum.org/>`_. MPI support can be enabled with ``PIKA_WITH_MPI=ON``.
+* `Boost.Context <https://boost.org>`_ on macOS or exotic platforms which are not supported by the
+  default user-level thread implementations in pika. This can be enabled with
+  ``PIKA_WITH_GENERIC_CONTEXT_COROUTINES=ON`` (TODO rename option...).
```

Comment:
Will this be renamed before or after the documentation is released?

Reply:
Unsure. We have to decide on a name first... I'll probably disentangle the renaming from this PR and do the renaming afterwards. I just wanted a reminder for myself to rename it sooner rather than later.

I'm currently thinking simply `PIKA_WITH_BOOST_CONTEXT` is the most descriptive name for it. Thoughts? Other suggestions?

Reply:
#729.